integration/system/event_test.go

```diff
@@ -58,3 +66,63 @@ func TestEvents(t *testing.T) {
 	}
 
 }
+
+func TestEventsEmptyOutput(t *testing.T) {
+	type apiResp struct {
+		resp *http.Response
+		err  error
+	}
+	chResp := make(chan *apiResp)
+	go func() {
+		resp, body, err := req.Get("/events")
+		body.Close()
+		chResp <- &apiResp{resp, err}
+	}()
+
+	select {
+	case r := <-chResp:
+		require.NoError(t, r.err)
+		assert.Equal(t, r.resp.StatusCode, http.StatusOK)
+	case <-time.After(3 * time.Second):
+		t.Fatal("timeout waiting for events api to respond, should have responded immediately")
+	}
+}
+
+func TestEventsBackwardsCompatible(t *testing.T) {
+	defer setupTest(t)()
+	ctx := context.Background()
+	client := request.NewAPIClient(t)
+
+	since := request.DaemonTime(ctx, t, client, testEnv)
+	ts := strconv.FormatInt(since.Unix(), 10)
+
+	cID := container.Run(t, ctx, client)
+	poll.WaitOn(t, container.IsInState(ctx, client, cID, "running"), poll.WithDelay(100*time.Millisecond))
```

Comment:
Wonder if we need the container to be running, or just created (as the test seems to only check for `created` events).

Just thinking out loud, we should verify if that doesn't change the intent of the test

Reply:
Thanks @thaJeztah, `Create` should be enough I think. The PR has been updated.